mono/metadata/metadata.c

```diff
@@ -2970,6 +2987,10 @@ mono_metadata_clean_for_image (MonoImage *image)
 		g_hash_table_foreach_steal (set->ginst_cache, steal_ginst_in_image, &ginst_data);
 		g_hash_table_foreach_remove (set->gmethod_cache, inflated_method_in_image, image);
 		g_hash_table_foreach_remove (set->gsignature_cache, inflated_signature_in_image, image);
+
+		g_hash_table_foreach_steal (set->ptr_cache, class_in_image, image);
```

Comment:
It looks like some tests (at least on the FullAOT lane) are failing with
```
ghashtable.c:509: assertion 'hash != NULL' failed
```
So these calls to `g_hash_table_foreach_steal` all need null-checks